### PR TITLE
postgresql_set: fix crashes caused by values containing 'b'

### PR DESCRIPTION
**Pull request by Andersson007**
##### SUMMARY
Fixes https://github.com/ansible-collections/community.postgresql/issues/48

I also made additional tests tests/integration/targets/postgresql_set/tasks/options_coverage.yml

##### ISSUE TYPE

- Bugfix Pull Request

##### COMPONENT NAME
postgresql_set

### CODE REVIEW

**Comment by Andersson007** [comment]
recheck

**Comment by Andersson007** [comment]
ready_for_review

**Comment by Andersson007** [comment]
ready_for_review

File: plugins/modules/postgresql_set.py [hunk 1]
```diff
@@ -240,33 +240,57 @@ def pretty_to_bytes(pretty_val):
...
-        val_in_bytes = num_part * 1024
+    if len(pretty_val) >= 2:
+        if 'kB' in pretty_val[-2:]:
+            val_in_bytes = num_part * 1024
 
-    elif 'MB' in pretty_val.upper():
-        num_part = int(''.join(d for d in pretty_val if d.isdigit()))
-        val_in_bytes = num_part * 1024 * 1024
+        elif 'MB' in pretty_val[-2:]:
+            val_in_bytes = num_part * 1024 * 1024
```
Comment (tcraxs) [inline plugins/modules/postgresql_set.py:277]: What about `mb`? Is this still covered like in line 249 with `pretty_val.upper()`?
Same is for `GB`, `TB` and `B`, please check for case sensitive.

Comment (Andersson007) [reply plugins/modules/postgresql_set.py:277]: @tcraxs yes, it should be covered, i consider this `.upper()` is extra because of line 345 in `main()`:
```
        for unit in POSSIBLE_SIZE_UNITS:
            if (value[:-2].isdigit() and unit in value[-2:]) or ('b' in value[-1] and value[:-1].isdigit()):
                value = value.upper()
```
and the function is called after this transformation so removed.
Also the CI tests cover the case you mentioned

Comment (tcraxs) [reply plugins/modules/postgresql_set.py:277]: Okay, perfect. I just oversaw this (also in the test cases).

**Review by tcraxs — APPROVED** [review]
LGTM

File: plugins/modules/postgresql_set.py (hunk 1 above)
Comment (Andersson007) [reply plugins/modules/postgresql_set.py:277]: np, I also oversaw this when writing the module;)

File: plugins/modules/postgresql_set.py [hunk 2]
```diff
@@ -182,7 +182,7 @@
 PG_REQ_VER = 90400
 
 # To allow to set value like 1mb instead of 1MB, etc:
-POSSIBLE_SIZE_UNITS = ("mb", "gb", "tb", "b")
+POSSIBLE_SIZE_UNITS = ("mb", "gb", "tb")
```
Comment (tcraxs) [inline plugins/modules/postgresql_set.py]: Is this intended? Why did you remove the `b` in this commit?

File: plugins/modules/postgresql_set.py [hunk 3]
```diff
@@ -240,33 +240,57 @@ def pretty_to_bytes(pretty_val):
...
+    # Extract digits
+    num_part = []
+    for c in pretty_val:
+        # When we reach the first non-digit element,
+        # e.g. in 1024kB, stop iterating
+        if not c.isdigit():
+            break
+
+        if c.isdigit():
+            num_part.append(c)
```
Comment (k3rni) [inline plugins/modules/postgresql_set.py:266]: conditions in lines 262 and 265 are inverses of each other. So why not just use an `else`?

File: plugins/modules/postgresql_set.py [hunk 4]
```diff
@@ -318,7 +342,7 @@ def main():
     # Allow to pass values like 1mb instead of 1MB, etc:
     if value:
         for unit in POSSIBLE_SIZE_UNITS:
-            if value[:-2].isdigit() and unit in value[-2:]:
+            if (value[:-2].isdigit() and unit in value[-2:]) or ('b' in value[-1] and value[:-1].isdigit()):
```
Comment (k3rni) [inline plugins/modules/postgresql_set.py:345]: this checks for the single `b` each time. shouldn't that check be outside of the for loop?

Comment (Andersson007) [reply plugins/modules/postgresql_set.py:345]: @k3rni yes, i'm in the business trip now, I'll rewrite the whole loop soon, I noticed that it's extra. Will fix it, thanks!

File: plugins/modules/postgresql_set.py (hunk 2 above)
Comment (Andersson007) [reply plugins/modules/postgresql_set.py]: @tcraxs sorry for late response, I'm in a business trip, I added it in this PR but it's wrong (I'll add an explanation to the comment to this tuple), I'll continue to work on the PR soon, will let you know.

File: plugins/modules/postgresql_set.py (hunk 3 above)
Comment (Andersson007) [reply plugins/modules/postgresql_set.py:265]: good spot! changed, thanks:)

File: plugins/modules/postgresql_set.py (hunk 4 above)
Comment (Andersson007) [reply plugins/modules/postgresql_set.py]: Rewrote. I could've written one long condition but split it intentionally for better readability

File: plugins/modules/postgresql_set.py (hunk 2 above)
Comment (Andersson007) [reply plugins/modules/postgresql_set.py]: I replaced the redundant loop where it's used, could you please take a look

**Comment by Andersson007** [comment]
ready_for_review

**Comment by tcraxs** [comment]
LGTM

**Comment by Andersson007** [comment]
@tcraxs @k3rni thanks for reviewing!
